### ISSUE TITLE
Transparency broken on X11

### ISSUE THREAD

**Issue by chrisduerr** [rust-windowing/glutin#1191]
Apparently since the 0.20 update to glutin (which bumped the major of winit), there have been some issues with transparency on the X11 backend.

The problem seems to boil down to `with_transparency(true)` acting like `with_transparency(false)`. So there are no crashes, but a fully transparent window is displayed as a black window.

The issue has been reported to Alacritty here: https://github.com/jwilm/alacritty/issues/2254

Other than X11 I haven't been able to find any significant similarities between the parties for which this works and for whom it doesn't. Not everyone on X11 is experiencing this issue.

The exact winit commit has not been bisected yet either, though that has already been requested.

**Comment by ZeGentzy** [comment]
As an update: currently the extension is being reviewed. https://bugs.freedesktop.org/show_bug.cgi?id=67676

**Comment by ZeGentzy** [comment]
![](https://imgur.com/WuhgZEl.png)

**Comment by ZeGentzy** [comment]
Merged into mesa! The patch should land in the next stable mesa release. Unfortunately landed one day too late to be included in Mesa 19.2.1 >.<

Expect to see this fixed in Mesa 19.2.2, which releases on 2019-10-23!

**Comment by okias** [comment]
Since the original MR got reverted, it's fixed in Mesa 24.1.

**Comment by kchibisov** [comment]
In the current glutin transparency is not blocked on EGL in any way iirc, so it should just work if it got fixed by mesa.
